Use UrlGeneratorInterface constants for UrlGenerator::generate

ahilles107 opened the pull request:
Don't use hardcoded values for the $referenceType argument of the Symfony\Component\Routing\Generator\UrlGenerator::generate

for full absolute path use Symfony\Component\Routing\Generator\UrlGeneratorInterface::ABSOLUTE_URL 
for relative path (with absolute url) use Symfony\Component\Routing\Generator\UrlGeneratorInterface::ABSOLUTE_PATH

ahilles107 commented:
Keeping BC will be hard as new constants in UrlGeneratorInterface were introduced in 2.8. So we should add version check for code and tests?

willdurand commented:
Yes..

ahilles107 commented:
not to myself. Possible solution: https://github.com/symfony-cmf/RoutingBundle/blob/master/Admin/RouteAdmin.php#L60

bendavies commented:
Need to update Travis to test multiple symfony versions too

bendavies commented:
do you need help with this?

ahilles107 commented:
No, just time - i will push something probably tomorrow. Sorry - Christmas is coming ;)

bendavies commented:
no problem!

ahilles107 commented:
@willdurand - like that it works in tests and in my project (based on Symfony 2.8)

willdurand commented:
good job! would you mind squash your commits please?

bendavies commented:
seems like the travis matrix needs changing to test all supported symfony versions.
see https://github.com/FriendsOfSymfony/FOSOAuthServerBundle/blob/master/.travis.yml for an example

ahilles107 commented:
I know, but this can go in other PR - no?

bendavies commented:
well that's up to @willdurand

willdurand commented:
either another commit or another PR, don't really mind

ahilles107 commented:
hey, i changed symfony coponents version in composer to ~2.4 and added direct dependecy on phpoption/phpoption with version ">=1.1.0,<2.0-dev" as there is a bug in schmittjoh/parser-lib  - https://github.com/schmittjoh/parser-lib/pull/8

willdurand commented:
I think it is fine since EL component is already required with `~2.4`, so :+1:

takeit commented:
:+1:

ahilles107 commented:
rebased

willdurand commented:
great, thanks!
